ale/drivers/base.py

```diff
@@ -942,7 +952,9 @@ def number_of_quaternions(self):
         int :
             The number of quaternions
         """
-        return len(self._sensor_orientation)
+        if not hasattr(self, "_inst_pointing_table"):
+            self._init_tables()
```

Comment:
For example here, just call `parse_position_table` if the table['Name'] == 'InstrumentPosition'. If 'InstrumentPosition' is not found, raise as appropriate. This saves a future dev having to wander over to the if/else method.

Reply:
All of the tables are read in at once to just have a single pass of file IO because the `_init_tables()` method opens the Cube file. It also give us a simpler object state; either everything is initialized or nothing is.

We could make it, so that it has separate methods for initializing the different tables but there would be a little bit of duplicated code and the potential to have things partially initialized.

Reply:
Keep a file handle around if I/O is a worry. Getting the data off disk is going to be meaningless compared to the time to get the handle I suspect. The data aren't that big. Then this can be separated into the methods that are calling it (my comment above). This serves two purposes - it gets rid of the above method and it gets the reads to be in the methods that are using the read data. 

Assuming all of the properties are using the hasattr paradigm, how could we get to a partial initialization?